src/Symfony/Component/PropertyInfo/WriteMutator.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * Writes mutator tell how to write to a property.
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+final class WriteMutator
```

Comment:
what about collection fields?

do we have type information about the property, or is that out of scope for this?

Reply:
Type information is out of scope, it should be extracted using a PropertyTypeExtractor

Reply:
but does the WriteMutator help with how to write to a collection field, or would that be up to the consumer to figure out? either there is add/remove or there is a property that is a collection that implements those methods... what if i should use a getter to get the collection and then modify it?

Reply:
This is up to the consumer to figure out the best way to do that